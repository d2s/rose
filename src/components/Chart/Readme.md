### Chart example

#### Default Chart (without props)

```js
<Chart />
```


#### Chart with a custom title (without defined content)

```js
<Chart title="Current Rate" />
```
