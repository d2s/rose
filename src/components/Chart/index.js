import React, { Component } from 'react';
// aimport ReactDOM from 'react-dom';
import { VictoryBar, VictoryChart, VictoryAxis, VictoryTheme } from 'victory';
import PropTypes from 'prop-types';
import glamorous, {ThemeProvider} from 'glamorous';

// Load required components from other directories


/**
 * Example data
 * 
 * - http://fixer.io/
 * - https://api.fixer.io/2018-01-24
 * - https://api.fixer.io/2018-01-30
 */
const data = [
  // {date: '2018-01-30', USD: 1.2421}
  {date: 1, USD: 1.2352},
  {date: 2, USD: 1.2407},
  {date: 3, USD: 1.2436},
  {date: 4, USD: 1.2436},
  {date: 5, USD: 1.2436},
  {date: 6, USD: 1.2379},
  {date: 7, USD: 1.2421}
];


// Chart content wrapper
const ChartWrapper = glamorous.div({
  backgroundColor: '#fff',
  color: '#000'
});


/**
 * Chart component for displaying chart data content.
 */
class Chart extends Component {
  static propTypes = {
    /** Text string for the Chart title, with a fallback defaultProps text if not specified. */
    title: PropTypes.string,
    /** Chart content. */
    content: PropTypes.any
  }
  static defaultProps = {
    title: 'Chart title',
    content: null
  }

  render() {
    return (
      <ChartWrapper>
        {/* {this.props.title} */}
        {/* {this.props.content} */}

        {/* // renders the default component with fallback data */}
        <VictoryChart
          domainPadding={30}
          theme={VictoryTheme.material}
          height={200}
        >
          <VictoryAxis
            tickValues={[1, 2, 3, 4, 5, 6, 7]}
            // tickValues={[1, 2, 3, 4, 5, 6, 7]}
            // tickFormat={["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]}
            // label="Weekday"
            padding={{ top: 20, bottom: 20 }}
            style={{
              ticks: {stroke: "grey", size: 5},
              tickLabels: {fontSize: 6, padding: 5}
            }}
          />
          <VictoryAxis
            dependentAxis
            tickFormat={(x) => (`$${x}`)}
            style={{
              ticks: {stroke: "grey", size: 5},
              tickLabels: {fontSize: 6, padding: 5}
            }}
          />
          <VictoryBar
            data={data}
            x="date"
            y="USD"
            labels={(d) => `${d.y}`}             
            barRatio={0.6}
            style={{
              labels: {
                fontSize: 7,
              },
            }}
          />
        </VictoryChart>
      </ChartWrapper>
    );
  }
}

export default Chart;
