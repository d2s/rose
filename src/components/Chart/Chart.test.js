import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

import Chart from './index';

describe('<Chart />', () => {
  describe('with ReactDOM', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Chart />, div);
      ReactDOM.unmountComponentAtNode(div);
    });
  });

  describe('with Enzyme', () => {
    it('shallow renders without crashing', () => {
      shallow(<Chart />);
    });
  });

  describe('with Snapshot Testing', () => {
    it('renders the same way', () => {
      const component = renderer.create(<Chart />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    });
  });
});
