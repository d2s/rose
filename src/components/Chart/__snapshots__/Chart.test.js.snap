// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart /> with Snapshot Testing renders the same way 1`] = `
<div
  className="css-1kgtxjy"
>
  <div
    className="VictoryContainer"
    style={
      Object {
        "height": "100%",
        "pointerEvents": "none",
        "position": "relative",
        "touchAction": "none",
        "userSelect": "none",
        "width": "100%",
      }
    }
  >
    <svg
      aria-labelledby="victory-container-2-title victory-container-2-desc"
      height={200}
      role="img"
      style={
        Object {
          "height": "100%",
          "pointerEvents": "all",
          "width": "100%",
        }
      }
      viewBox="0 0 350 200"
      width={350}
    >
      <g
        role="presentation"
      >
        <line
          className={undefined}
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "transparent",
              "stroke": "#90A4AE",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 2,
            }
          }
          vectorEffect="non-scaling-stroke"
          x1={50}
          x2={300}
          y1={150}
          y2={150}
        />
        <g
          role="presentation"
        >
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "visible",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={78.66831072749692}
            x2={78.66831072749692}
            y1={150}
            y2={50}
          />
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "grey",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={78.66831072749692}
            x2={78.66831072749692}
            y1={150}
            y2={155}
          />
          <text
            className={undefined}
            dx={0}
            dy={5.13}
            transform={undefined}
            x={78.66831072749692}
            y={160}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 6,
                  "letterSpacing": "normal",
                  "padding": 5,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={78.66831072749692}
            >
              1
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "visible",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={110.77887381833128}
            x2={110.77887381833128}
            y1={150}
            y2={50}
          />
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "grey",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={110.77887381833128}
            x2={110.77887381833128}
            y1={150}
            y2={155}
          />
          <text
            className={undefined}
            dx={0}
            dy={5.13}
            transform={undefined}
            x={110.77887381833128}
            y={160}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 6,
                  "letterSpacing": "normal",
                  "padding": 5,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={110.77887381833128}
            >
              2
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "visible",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={142.88943690916562}
            x2={142.88943690916562}
            y1={150}
            y2={50}
          />
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "grey",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={142.88943690916562}
            x2={142.88943690916562}
            y1={150}
            y2={155}
          />
          <text
            className={undefined}
            dx={0}
            dy={5.13}
            transform={undefined}
            x={142.88943690916562}
            y={160}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 6,
                  "letterSpacing": "normal",
                  "padding": 5,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={142.88943690916562}
            >
              3
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "visible",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={175}
            x2={175}
            y1={150}
            y2={50}
          />
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "grey",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={175}
            x2={175}
            y1={150}
            y2={155}
          />
          <text
            className={undefined}
            dx={0}
            dy={5.13}
            transform={undefined}
            x={175}
            y={160}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 6,
                  "letterSpacing": "normal",
                  "padding": 5,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={175}
            >
              4
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "visible",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={207.11056309083435}
            x2={207.11056309083435}
            y1={150}
            y2={50}
          />
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "grey",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={207.11056309083435}
            x2={207.11056309083435}
            y1={150}
            y2={155}
          />
          <text
            className={undefined}
            dx={0}
            dy={5.13}
            transform={undefined}
            x={207.11056309083435}
            y={160}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 6,
                  "letterSpacing": "normal",
                  "padding": 5,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={207.11056309083435}
            >
              5
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "visible",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={239.22112618166872}
            x2={239.22112618166872}
            y1={150}
            y2={50}
          />
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "grey",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={239.22112618166872}
            x2={239.22112618166872}
            y1={150}
            y2={155}
          />
          <text
            className={undefined}
            dx={0}
            dy={5.13}
            transform={undefined}
            x={239.22112618166872}
            y={160}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 6,
                  "letterSpacing": "normal",
                  "padding": 5,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={239.22112618166872}
            >
              6
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "visible",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={271.33168927250307}
            x2={271.33168927250307}
            y1={150}
            y2={50}
          />
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "grey",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={271.33168927250307}
            x2={271.33168927250307}
            y1={150}
            y2={155}
          />
          <text
            className={undefined}
            dx={0}
            dy={5.13}
            transform={undefined}
            x={271.33168927250307}
            y={160}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 6,
                  "letterSpacing": "normal",
                  "padding": 5,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={271.33168927250307}
            >
              7
            </tspan>
          </text>
        </g>
      </g>
      <g
        role="presentation"
      >
        <line
          className={undefined}
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "transparent",
              "stroke": "#90A4AE",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 2,
            }
          }
          vectorEffect="non-scaling-stroke"
          x1={50}
          x2={50}
          y1={50}
          y2={150}
        />
        <g
          role="presentation"
        >
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "visible",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={50}
            x2={300}
            y1={121.07492519975656}
            y2={121.07492519975656}
          />
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "grey",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={50}
            x2={45}
            y1={121.07492519975656}
            y2={121.07492519975656}
          />
          <text
            className={undefined}
            dx={0}
            dy={2.13}
            transform={undefined}
            x={40}
            y={121.07492519975656}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 6,
                  "letterSpacing": "normal",
                  "padding": 5,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={40}
            >
              $0.5
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "visible",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={50}
            x2={300}
            y1={92.14985039951313}
            y2={92.14985039951313}
          />
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "grey",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={50}
            x2={45}
            y1={92.14985039951313}
            y2={92.14985039951313}
          />
          <text
            className={undefined}
            dx={0}
            dy={2.13}
            transform={undefined}
            x={40}
            y={92.14985039951313}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 6,
                  "letterSpacing": "normal",
                  "padding": 5,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={40}
            >
              $1
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "visible",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={50}
            x2={300}
            y1={63.224775599269705}
            y2={63.224775599269705}
          />
          <line
            className={undefined}
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "grey",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={50}
            x2={45}
            y1={63.224775599269705}
            y2={63.224775599269705}
          />
          <text
            className={undefined}
            dx={0}
            dy={2.13}
            transform={undefined}
            x={40}
            y={63.224775599269705}
          >
            <tspan
              dx={0}
              dy={undefined}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 6,
                  "letterSpacing": "normal",
                  "padding": 5,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={40}
            >
              $1.5
            </tspan>
          </text>
        </g>
      </g>
      <g
        role="presentation"
      >
        <path
          className={undefined}
          d="M 70.33497739416359, 150
              L 70.33497739416359, 78.54349521347862
              A 0 0 0 0 1, 70.33497739416359, 78.54349521347862
              L 87.00164406083024, 78.54349521347862
              A 0 0 0 0 1, 87.00164406083024, 78.54349521347862
              L 87.00164406083024, 150
              L 70.33497739416359, 150
              z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "USD": 1.2352,
              "date": 1,
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
          transform={undefined}
        />
        <path
          className={undefined}
          d="M 102.44554048499795, 150
              L 102.44554048499795, 78.22531939067595
              A 0 0 0 0 1, 102.44554048499795, 78.22531939067595
              L 119.1122071516646, 78.22531939067595
              A 0 0 0 0 1, 119.1122071516646, 78.22531939067595
              L 119.1122071516646, 150
              L 102.44554048499795, 150
              z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "USD": 1.2407,
              "date": 2,
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
          transform={undefined}
        />
        <path
          className={undefined}
          d="M 134.55610357583228, 150
              L 134.55610357583228, 78.05755395683454
              A 0 0 0 0 1, 134.55610357583228, 78.05755395683454
              L 151.22277024249897, 78.05755395683454
              A 0 0 0 0 1, 151.22277024249897, 78.05755395683454
              L 151.22277024249897, 150
              L 134.55610357583228, 150
              z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "USD": 1.2436,
              "date": 3,
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
          transform={undefined}
        />
        <path
          className={undefined}
          d="M 166.66666666666666, 150
              L 166.66666666666666, 78.05755395683454
              A 0 0 0 0 1, 166.66666666666666, 78.05755395683454
              L 183.33333333333334, 78.05755395683454
              A 0 0 0 0 1, 183.33333333333334, 78.05755395683454
              L 183.33333333333334, 150
              L 166.66666666666666, 150
              z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "USD": 1.2436,
              "date": 4,
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
          transform={undefined}
        />
        <path
          className={undefined}
          d="M 198.777229757501, 150
              L 198.777229757501, 78.05755395683454
              A 0 0 0 0 1, 198.777229757501, 78.05755395683454
              L 215.4438964241677, 78.05755395683454
              A 0 0 0 0 1, 215.4438964241677, 78.05755395683454
              L 215.4438964241677, 150
              L 198.777229757501, 150
              z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "USD": 1.2436,
              "date": 5,
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
          transform={undefined}
        />
        <path
          className={undefined}
          d="M 230.88779284833538, 150
              L 230.88779284833538, 78.3872998095573
              A 0 0 0 0 1, 230.88779284833538, 78.3872998095573
              L 247.55445951500207, 78.3872998095573
              A 0 0 0 0 1, 247.55445951500207, 78.3872998095573
              L 247.55445951500207, 150
              L 230.88779284833538, 150
              z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "USD": 1.2379,
              "date": 6,
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
          transform={undefined}
        />
        <path
          className={undefined}
          d="M 262.99835593916976, 150
              L 262.99835593916976, 78.14432918123526
              A 0 0 0 0 1, 262.99835593916976, 78.14432918123526
              L 279.6650226058364, 78.14432918123526
              A 0 0 0 0 1, 279.6650226058364, 78.14432918123526
              L 279.6650226058364, 150
              L 262.99835593916976, 150
              z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "USD": 1.2421,
              "date": 7,
              "fill": "#455A64",
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 0,
            }
          }
          transform={undefined}
        />
        <text
          className={undefined}
          dx={0}
          dy={-1.0150000000000001}
          transform={undefined}
          x={78.66831072749692}
          y={70.54349521347862}
        >
          <tspan
            dx={0}
            dy={undefined}
            style={
              Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 7,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              }
            }
            textAnchor="middle"
            x={78.66831072749692}
          >
            1.2352
          </tspan>
        </text>
        <text
          className={undefined}
          dx={0}
          dy={-1.0150000000000001}
          transform={undefined}
          x={110.77887381833128}
          y={70.22531939067595}
        >
          <tspan
            dx={0}
            dy={undefined}
            style={
              Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 7,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              }
            }
            textAnchor="middle"
            x={110.77887381833128}
          >
            1.2407
          </tspan>
        </text>
        <text
          className={undefined}
          dx={0}
          dy={-1.0150000000000001}
          transform={undefined}
          x={142.88943690916562}
          y={70.05755395683454}
        >
          <tspan
            dx={0}
            dy={undefined}
            style={
              Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 7,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              }
            }
            textAnchor="middle"
            x={142.88943690916562}
          >
            1.2436
          </tspan>
        </text>
        <text
          className={undefined}
          dx={0}
          dy={-1.0150000000000001}
          transform={undefined}
          x={175}
          y={70.05755395683454}
        >
          <tspan
            dx={0}
            dy={undefined}
            style={
              Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 7,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              }
            }
            textAnchor="middle"
            x={175}
          >
            1.2436
          </tspan>
        </text>
        <text
          className={undefined}
          dx={0}
          dy={-1.0150000000000001}
          transform={undefined}
          x={207.11056309083435}
          y={70.05755395683454}
        >
          <tspan
            dx={0}
            dy={undefined}
            style={
              Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 7,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              }
            }
            textAnchor="middle"
            x={207.11056309083435}
          >
            1.2436
          </tspan>
        </text>
        <text
          className={undefined}
          dx={0}
          dy={-1.0150000000000001}
          transform={undefined}
          x={239.22112618166872}
          y={70.3872998095573}
        >
          <tspan
            dx={0}
            dy={undefined}
            style={
              Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 7,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              }
            }
            textAnchor="middle"
            x={239.22112618166872}
          >
            1.2379
          </tspan>
        </text>
        <text
          className={undefined}
          dx={0}
          dy={-1.0150000000000001}
          transform={undefined}
          x={271.33168927250307}
          y={70.14432918123526}
        >
          <tspan
            dx={0}
            dy={undefined}
            style={
              Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 7,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              }
            }
            textAnchor="middle"
            x={271.33168927250307}
          >
            1.2421
          </tspan>
        </text>
      </g>
    </svg>
    <div
      style={
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": 99,
        }
      }
    >
      <svg
        height={200}
        style={
          Object {
            "height": "100%",
            "overflow": "visible",
            "width": "100%",
          }
        }
        viewBox="0 0 350 200"
        width={350}
      />
    </div>
  </div>
</div>
`;
